### ISSUE TITLE
Out of range error using staged_predict

### ISSUE THREAD

**Issue by ghost** [catboost/catboost#1310]
Problem: Hello all, I am trying to use the staged_predict method on my CatBoostClassifier (parameters are {'loss_function': 'Logloss', 'depth': 7, 'iterations': 1000, 'learning_rate': 0.1}).
I wrote : 
Z = model.staged_predict(data=X_train, ntree_start=0, ntree_end=40)
A = list(Z)
But got : catboost/private/libs/algo/apply.cpp:41: Out of range treeEnd=26
How can I fix that error ? 
Thanks !

catboost version: 0.23.2
Operating System: Mac OS
CPU:
# GPU:

**Comment by andrey-khropov** [comment]
How many trees are really in your model (you can check from the training logs or [`tree_count_` attribute](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier_attributes#tree_count_) )? Even if you set `iterations` to `1000` the real number of trees can be lower when [`use_best_model`](https://catboost.ai/en/docs/concepts/python-reference_catboostclassifier_fit#use_best_model) is enabled (true by default if `eval_set` is provided) and the number of trees is chosen based on the best `eval_metric` value on the validation set.

I suspect that this is the issue - your model does in fact have less than 26 trees.

**Comment by andrey-khropov** [comment]
Actually the author of the issue is now @ghost, so he/she can't provide any feedback.
